server/src/test/java/com/broadcom/lsp/cobol/usecases/TestFormatTrim.java

```diff
@@ -34,33 +35,51 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
 
-class TestFormatTrim extends ConfigurableTest {
+class TestFormatTrim extends MockTextDocumentService {
   private static final String ID = "id";
-
-  private TextDocumentService service;
-  private TestLanguageClient client;
-
   private static final Pattern RTRIM = Pattern.compile("\\s+$");
-
   private static final String TEXT =
       "000000 Identification DIVISION.                                         23323232\r\n"
           + "002800 Program-ID.                                                      23323232  \r\n"
           + "002800  HELLOWORLD.                                                     233232322@\t                                          \r\n"
           + "223d3& AUTHOR.\r\n"
           + "002800 INPUT-OUTPUT SECTION.                                            23323232  ";
+  private TextDocumentService service;
+
+  private static String rtrim(String s) {
+    return RTRIM.matcher(s).replaceAll("");
+  }
 
   @BeforeEach
   void createService() {
-    service = injector.getInstance(TextDocumentService.class);
-    client = injector.getInstance(TestLanguageClient.class);
-    client.clean();
+    service = getMockedTextDocumentService();
+    when(engine.analyze(anyString(), anyString(), any(CopybookProcessingMode.class)))
+        .thenReturn(AnalysisResult.empty());
     service.didOpen(new DidOpenTextDocumentParams(new TextDocumentItem(ID, "COBOL", 1, TEXT)));
   }
 
   @Test
   void testTrimming() {
-    UseCaseUtils.waitForDiagnostics(client);
+    when(formations.format(any(CobolDocumentModel.class)))
```

Comment:
This test should check the actual `TrimFormation` behavior. 
It was done incorrectly from the beginning, and it would be better to split it into unit tests for `TrimFormation`, `Formation`, and one more test method in `CobolTextDocumentServiceTest` that checks that the `formatting` calls the `Formations` delegate